search-sentiments-map/routes/search.js

```diff
@@ -0,0 +1,278 @@
+      console.log(`Custom Search Result ${searchResultKey.id} deleted.`)
+    } else {
+      break;
+    }
+  }
+}
+
+/** 
+ * Creates Datastore item for given topic and country search results. 
+ * @param {string} topic The seach topic the search results are for.
+ * @param {Object} countriesData Object holding all searchResults for all
+ *      countries.
+ */
+async function addTopicToDatastore(topic, countriesData) {
+  const customSearchTopicKey = datastore.key('CustomSearchTopic');
+  // Get the current timestamp in milliseconds.
+  let timestamp = Date.now();
+  const entity = {
+    key: customSearchTopicKey,
+    data: {
+      topic: topic,
+      countries: countriesData,
+      timestamp: Date.now(),
+    },
+  };
+  try {
+    await datastore.save(entity);
+    console.log(`Custom Search Result ${customSearchTopicKey.id} created successfully.`);
+  } catch (err) {
+    console.error('ERROR:', err);
+  }
+}
+
+/**
+ * Adds search result object to countryData array.
+ * @param {Object} searchResult Object with information for one search result.
+ * @param {Object} countryData Object holding all searchResults for a country.
+ * @return {Promise} A Promise wrapped around the sentiment score of the result.
+*/
+function addSearchResultToCountryData(searchResult, countryData) {
+  return getSentiment(searchResult)
+    .then(response => response.json())
+    .then((result) => {
+      searchResultData = {
+        title: searchResult.title,
+        snippet: searchResult.snippet,
+        htmlTitle: searchResult.htmlTitle,
+        link: searchResult.link,
+        score: result.score,
+      };
+      countryData.push(searchResultData);
+      return result.score;
+    });
+
+}
+
+// TODO(ntarn): Look into changing the above method so that it does not need to return 
+// twice for a Promise wrapped around a object.
+// async function addSearchResultToCountryData(searchResult, countryData) {
+//   let sentimentScore;
+//   await getSentiment(searchResult)
+//     .then(response => response.json())
+//     .then((result) => { 
+//       console.log('ntarn debug: frontend' + result.score);
+//       searchResultData = {
+//         title: searchResult.title,
+//         snippet: searchResult.snippet,
+//         htmlTitle: searchResult.htmlTitle,
+//         link: searchResult.link,
+//         score: result.score,
+//       };
+//       countryData.push(searchResultData);
+//       sentimentScore = result.score;
+//     });
+//   return sentimentScore;
+// }
+
+/** 
+ * Gets the sentiment score of a search result. 
+ * @param {Object} searchResult Object for one search result.
+ */
+function getSentiment(searchResult) {
+  return fetch('https://trending-search-sentiments.ue.r.appspot.com/sentiment', {
+    method: 'POST',  // Send a request to the URL.
+    headers: new Headers({
+      'Content-Type': 'text/plain',
+    }),
+    body: searchResult.title + searchResult.snippet
+  })
+    .catch(err => {
```

Comment:
This should be 4 space indented.

Reply:
maybe append catch on to the previous line